tests/tree_util_test.py

```diff
@@ -1294,6 +1294,36 @@ def test_tree_unflatten(self):
 
 class RegistrationTest(jtu.JaxTestCase):
 
+  def test_register_dataclass_with_field_specifier(self):
+    @tree_util.register_dataclass
+    @dataclasses.dataclass
+    class Foo:
+      x: int
+      y: int = tree_util.field(static=True)
```

Comment:
This looks okay to me, but I wonder if we should instead tell people to just do `dataclasses.field(metadata=dict(static=True))`?

Reply:
It's a bit of a mouthful, but it would be a more incremental change and not require a new API... maybe better to go with that for this change.